specs/gateway.yaml

```diff
@@ -0,0 +1,400 @@
+      name:
+        type: string
+      faceImageUrl:
+        type: string
+
+  Villain:
+    type: object
+    description: A description of an enemy in the world space.
+    properties:
+      id:
+        type: string
+        description: Primary key that refers to this instance of this villan.
+      villainTemplateId:
+        type: string
+      stats:
+        $ref: '#/definitions/Stats'
+      attributes:
+        type: array
+        description: List of abilities this villan has.
+        items:
+          $ref: '#/definitions/Attribute'
+      location:
+        $ref: '#/definitions/Location'
+        description: Where in the world space
+
+  ItemPickup:
+    type: object
+    description: Stack of items that can be picked up in the world space
+    properties:
+      itemStack:
+        $ref: '#/definitions/ItemStack'
+      location:
+        $ref: '#/definitions/Location'
+
+  Room:
+    type: object
+    description: Properties of a room
+    properties:
+      id:
+        type: string
+        description: ID of the room
+      playerId:
+        type: string
+        description: Player that owns this room
+      weatherModifier:
+        type: string
+        description: >
+          How to modify the atmosphere of the room based on the weather of the player at time
+          it was created. This is used to add extra flavor/emotion to the game.
+      villans:
+        type: array
+        items:
+          $ref: '#/definitions/Villain'
+      pickups:
+        type: array
+        items:
+          $ref: '#/definitions/ItemPickup'
+      tiles:
+        type: array
+        items:
+          $ref: '#/definitions/Tile'
+          minimum: 10000
+          maximum: 10000 # TileXMaximum * TileYMaximum
+
+
+  Inventory:
+    type: object
+    properties:
+      id:
+        type: string
+        description: ID of this inventory
+      playerId:
+        type: string
+        description: Player that owns this inventory.
+      itemStacks:
+        type: array
+        items:
+          $ref: '#/definitions/ItemStack'
+
+  Login:
+    type: object
+    description: Data used to log in to an account
+    properties:
+      username:
+        type: string
+      password:
+        type: string
+        minLength: 0
+        maxLength: 127
+
+  PlayerCreation:
+    type: object
+    additionalProperties:
+      $ref: '#/definitions/Login'
+    properties:
+      locationName:
+        type: string
+        description: Name of the location the user is in
+
+  Player:
+    type: object
+    properties:
+      id:
+        type: string
+        description: ID of this player
+      username:
+        type: string
+        description: Username of the player
+      inventoryId:
+        type: string
+        description: ID of the player's inventory
+      roomId:
+        type: string
+        description: Room the player is in. Null if player is not in a room.
+      stats:
+        $ref: '#/definitions/Stats'
```

Comment:
Add abilities property, type: array, description: List of abilities the player has (I'll need to make a list of abilities)

Reply:
Added into `Stats` object.